packages/@aws-cdk/aws-secretsmanager/lib/rotation-schedule.ts

```diff
@@ -39,12 +49,364 @@ export class RotationSchedule extends Resource {
+      ...options,
+      type: HostedRotationType.REDSHIFT_SINGLE_USER,
+    });
+  }
+
+  /**
+   * Redshift Multi User
+   */
+  public static redshiftMultiUser(options: MultiUserHostedRotationOptions) {
+    return new HostedRotation({
+      ...options,
+      type: HostedRotationType.REDSHIFT_MULTI_USER,
+    });
+  }
+
+  /**
+   * MongoDB Single User
+   */
+  public static mongoDbSingleUser(options: SingleUserHostedRotationOptions = {}) {
+    return new HostedRotation({
+      ...options,
+      type: HostedRotationType.MONGODB_SINGLE_USER,
+    });
+  }
+
+  /**
+   * MongoDB Multi User
+   */
+  public static mongoDbMultiUser(options: MultiUserHostedRotationOptions) {
+    return new HostedRotation({
+      ...options,
+      type: HostedRotationType.MONGODB_MULTI_USER,
+    });
+  }
+
+  private securityGroups?: ec2.ISecurityGroup[];
+
+  constructor(private readonly props: HostedRotationProps) {
+    if (props.type.isMultiUser && !props.masterSecret) {
+      throw new Error('The `masterSecret` must be specified when using the multi user scheme.');
+    }
+
+    this.securityGroups = this.props.securityGroups;
+  }
+
+  /**
+   * Binds this hosted rotation to a secret
+   */
+  public bind(secret: ISecret, scope: Construct): CfnRotationSchedule.HostedRotationLambdaProperty {
+    // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-secretsmanager-rotationschedule-hostedrotationlambda.html
+    Stack.of(scope).addTransform('AWS::SecretsManager-2020-07-23');
+
+    if (!this.props.vpc && this.props.securityGroups) {
+      throw new Error('`vpc` must be specified when specifying `securityGroups`.');
+    }
+
+    if (this.props.vpc && !this.props.securityGroups) {
+      this.securityGroups = [new ec2.SecurityGroup(scope, 'SecurityGroup', {
+        vpc: this.props.vpc,
+      })];
+    }
+
+    // Prevent master secret deletion when rotation is in place
+    if (this.props.masterSecret) {
+      this.props.masterSecret.denyAccountRootDelete();
+    }
+
+    return {
+      rotationType: this.props.type.name,
+      kmsKeyArn: secret.encryptionKey?.keyArn,
+      masterSecretArn: this.props.masterSecret?.secretArn,
+      masterSecretKmsKeyArn: this.props.masterSecret?.encryptionKey?.keyArn,
+      rotationLambdaName: this.props.functionName,
+      vpcSecurityGroupIds: this.securityGroups?.map(s => s.securityGroupId).join(','),
+      vpcSubnetIds: this.props.vpc?.selectSubnets(this.props.vpcSubnets).subnetIds.join(','),
+    };
+  }
+
+  /**
+   * Security group connections for this hosted rotation
+   */
+  public get connections() {
+    if (!this.props.vpc) {
+      throw new Error('Cannot use connections for a hosted rotation that is not deployed in a VPC');
+    }
+
+    if (!this.securityGroups) {
+      throw new Error('Cannot use connections for a hosted rotation that has not been bound to a secret');
+    }
+
+    return new ec2.Connections({ securityGroups: this.securityGroups });
```

Comment:
This will create a new object every time `connections()` is called; will be fixed if you store the `Connections`, rather than the security groups.